File: labs/lab1.1/cpp/main.cpp
/*
    Done by:
    Student: Hordiienko Volodymyr
    Group: 123:2
    Lab 1.1
    Variant 4
*/

#include "iostream"
#include "rectangle.cpp"
#include "vector2.cpp"

using std::cin;
using std::cout;

int main() {
    double x,y,width,height;
    cout << "Enter the X coord of the top left vertex: ";
    cin >> x;
    cout << "Enter the Y coord of the top left vertex: ";
    cin >> y;
    cout << "Enther the width of the rectangle: ";
    cin >> width;
    cout << "Enther the height of the rectangle: ";
    cin >> height;

    rectangle rect(vector2(x,y),width,height);

    cout << "Top left: " << rect.TopLeft().ToString() << '\n';
    cout << "Top right: " << rect.TopRight().ToString() << '\n';
    cout << "Bottom left: " << rect.BottomLeft().ToString() << '\n';
    cout << "Bottom right: " << rect.BottomRight().ToString() << '\n';
    cout << "Perimeter: " << rect.GetPerimeter() << '\n';
    cout << "Area: " << rect.GetArea() << '\n';

    return 0;
}

File: labs/lab1.1/typescript/main.ts
import { Vector2 } from "./vector2"
import { Rectange } from "./rectangle"
import { question } from "readline-sync"

function getInputAsNumber(query:string):number {
    let input = NaN
    while(true) {
        input = Number(question(query))
        if(isNaN(input)) console.log("Enter a valid number!")
        else return input
    }
}

const topLeftX = getInputAsNumber("Enter the X coord of the top left vertex: ")
const topLeftY = getInputAsNumber("Enter the Y coord of the top left vertex: ")
const width = getInputAsNumber("Enter the width of the rectangle: ")
const height = getInputAsNumber("Enter the height of the rectangle: ")

const rect = new Rectange(new Vector2(topLeftX,topLeftY),width,height)

console.log(`Top left: ${rect.TopLeft.ToString()}`)
console.log(`Top right: ${rect.TopRight.ToString()}`)
console.log(`Bottom left: ${rect.BottomLeft.ToString()}`)
console.log(`Bottomg right ${rect.BottomRight.ToString()}`)
console.log(`Perimeter: ${rect.GetPerimeter()}`)
console.log(`Area: ${rect.GetArea()}`)